tidy3d/plugins/microwave/path_integrals.py

```diff
@@ -0,0 +1,299 @@
+"""Helper classes for performing path integrals with fields on the Yee grid"""
+
+from __future__ import annotations
+
+import pydantic.v1 as pd
+import numpy as np
+from typing import Tuple
+from ...components.data.dataset import FieldDataset, FieldTimeDataset, ModeSolverDataset
+from ...components.data.data_array import AbstractSpatialDataArray
+from ...components.base import cached_property
+from ...components.types import Axis, Direction
+from ...components.geometry.base import Box
+from ...components.validators import assert_line, assert_plane
+from ...exceptions import DataError
+
+
+class AbstractAxesRH:
```

Comment:
the convenetion is to inherit abstract classes from `abc.ABC`. I think basically this just makes it not possible to initialize this class, with a useful error message.

Also, mightt want to consider making it inherit from `Tidy3dBaseModel` as well?